src/win/fs.c

```diff
@@ -2627,8 +2627,86 @@ static void fs__statfs(uv_fs_t* req) {
                              &bytes_per_sector,
                              &free_clusters,
                              &total_clusters)) {
-    SET_REQ_WIN32_ERROR(req, GetLastError());
-    return;
+    DWORD err;
+
+    err = GetLastError();
+    if (err == ERROR_DIRECTORY) {
+      WCHAR* pathw;
+      size_t len;
+      DWORD ret;
+
+      len = MAX_PATH + 1;
+      pathw = uv__malloc(len * sizeof(*pathw));
+retry:
+      ret = GetFullPathNameW(req->file.pathw,
+                             len,
+                             pathw,
+                             NULL);
+      if (ret == 0) {
+        uv__free(pathw);
+        SET_REQ_WIN32_ERROR(req, err);
+        return;
+      } else if (ret > len) {
```

Comment:
Can this actually happen? `len` starts out as `MAX_PATH + 1`, right?

Reply:
`GetFullPathNameW` can handle paths longer than `MAX_PATH` characters. Therefore, it is possible if such a path is given in the UNC path or if it is allowed by configuration on Windows 10, version 1607 or later.

Ref. https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getfullpathnamea